src/app/cli/tests/tests/test_di_graph.rs

```diff
@@ -0,0 +1,29 @@
+// Copyright Kamu Data, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use dill::*;
+use kamu::WorkspaceLayout;
+use kamu_cli;
+
+#[test_log::test(tokio::test)]
+async fn test_di_graph_validates() {
+    let mut catalog_builder = kamu_cli::configure_catalog();
+    kamu_cli::register_config_in_catalog(&kamu_cli::CLIConfig::default(), &mut catalog_builder);
+
+    // TODO: We should ensure this test covers parameters requested by commands and
+    // types needed for GQL/HTTP adapter that are currently being constructed
+    // manually
+    let validate_result = catalog_builder.validate().ignore::<WorkspaceLayout>();
```

Comment:
Notice: `dill` now has a feature that will statically validate the dependency graph to avoid most runtime errors.

Because GraphQL and HTTP adapters are using `Catalog` manually to create objects - they are still exposed to runtime errors. We will need to find a way to create a `dill-axum` and `dill-async-graphql` adapters to be able to statically analyze the dependencies of their endpoints and include them into this validation.

Reply:
For every registered `Builder` it will check that all of its dependencies can be satisfied. See: https://github.com/sergiimk/dill-rs/blob/master/dill/tests/tests/test_validation.rs